src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs

```diff
@@ -70,7 +70,7 @@ public HubConnection(IConnection connection, IInvocationAdapter adapter, ILogger
 
             _connection = connection;
             _binder = new HubBinder(this);
-            _adapter = adapter;
+            _protocol = protocol;
```

Comment:
I'd rather check `protocol` for null, since we have overloads that set the default value.